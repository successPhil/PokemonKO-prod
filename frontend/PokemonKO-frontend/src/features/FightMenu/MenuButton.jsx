import React from 'react';

const MenuButton = ({ label, onClick }) => {
  const buttonStyle = {
    cursor: 'pointer',
    border: 'none',
    background: 'transparent',
    fontFamily: 'Pokemon GB', // Apply your custom font
    fontSize: '1.7rem', // Adjust as needed
    color: '000000', // Adjust color to match your design
  };

  return (
    <button style={buttonStyle} onClick={onClick}>
      {label}
    </button>
  );
};

export default MenuButton;
