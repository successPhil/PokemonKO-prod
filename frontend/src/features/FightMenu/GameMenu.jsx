import Pokeball from "../../styles/Pokeball";
import MenuButton from "./MenuButton";
import { trainerRun } from "../../api/authApi";
import { useNavigate } from "react-router-dom";
import { useContext } from "react";
import TrainerContext from "../../contexts/TrainerContext";

export default function GameMenu({moves, openMoves, toggleMenu}) {
  const { setEnemyPokemon } = useContext(TrainerContext)

  const navigate = useNavigate()

  const handleFightClick = () => {
    console.log('click')
    toggleMenu(openMoves)
  }

  const navToItems = () => {
    console.log('clicked')
    return navigate("/items")
  }

  const handleRun = () => {
    trainerRun()
    setEnemyPokemon(null)

  }

  return (
    <div className='gameboy-menu-container'>
      <div className='gameboy-menu-box'>
        <div className='menu-content'>
        <div className="button-row">
        <MenuButton label="FIGHT" onClick={(handleFightClick)}  />
      <MenuButton label="POKE"  />
      </div>
      <div className="button-row">
      <MenuButton label="ITEMS" onClick={(navToItems)}  />
      <MenuButton label="RUN" onClick={handleRun}  />
      </div>
        </div>
      </div>
      <div className="top-left">
        <Pokeball />
      </div>
      <div className="top-right">
        <Pokeball />
      </div>
      <div className="bottom-left">
        <Pokeball />
      </div>
      <div className="bottom-right">
        <Pokeball />
      </div>
    </div>
  );
}
