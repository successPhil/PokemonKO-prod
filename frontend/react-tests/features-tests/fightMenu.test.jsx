import GameMenu from '../../src/features/FightMenu/GameMenu'
import MenuButton from '../../src/features/FightMenu/MenuButton'
import {expect, test} from 'vitest'

test("FightMenu components render without crashing", ()=> {
    const gameMenu = <GameMenu/>
    expect(gameMenu).toMatchSnapshot()
    const menuButton = <MenuButton/>
    expect(menuButton).toMatchSnapshot()
    
})