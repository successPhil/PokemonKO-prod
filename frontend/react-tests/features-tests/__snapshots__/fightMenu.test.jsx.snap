// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FightMenu components render without crashing 1`] = `<GameMenu />`;

exports[`FightMenu components render without crashing 2`] = `<MenuButton />`;
